test/test_facebook.py

```diff
@@ -190,6 +190,25 @@ def test_extend_access_token(self):
             self.assertEqual(
                 e.message, "fb_exchange_token parameter not specified")
 
+    def test_get_code_from_token(self):
+        """
+        Test that get_code_from_token returns a valid code which can be
+        exchanged for an access token.
+        """
+        redirect_uri = 'https://localhost/facebook/callback/'
+        app_token = facebook.GraphAPI().get_app_access_token(self.app_id,
+                                                             self.secret)
+        self.create_test_users(self.app_id,
+                          facebook.GraphAPI(app_token), 1)
+        test_token = self.test_users[0]['access_token']
+        graph = facebook.GraphAPI(test_token)
+        self.assertIsNotNone(graph.get_code_from_token(
+                        self.app_id, self.secret, redirect_uri),
+                        'Code not returned by get_code_from_token method')
+        self.assertIsNotNone(graph.get_access_token_from_code(
+                        self.app_id, self.secret), 'Invalid code returned by'
+                        'get_code_from_token method')
+
```

Comment:
Why does this test call both `get_code_from_token` and `get_access_token_from_code`? Should it be separated into two different tests?

I don't think it's safe to use `assertIsNotNone` here - this assumes that either method has succeeded as long as it doesn't return None. I think it would be better to test failure as well (e.g. using an expired token or a bad redirect URI).